pkg/operation/seed/seed.go

```diff
@@ -298,9 +325,147 @@ func BootstrapCluster(seed *Seed, k8sGardenClient kubernetes.Client, secrets map
 				"fluent-bit": fluentBitVersion.String(),
 			},
 		},
+		"cert-manager": map[string]interface{}{
+			"enabled": certManagerEnabled,
+			"images": map[string]interface{}{
+				"cert-manager": certManagerVersion.String(),
+			},
+			"clusterissuer": clusterIssuer,
+		},
 	})
 }
 
+func createClusterIssuer(k8sSeedclient kubernetes.Client, certificateManagement *corev1.Secret) (map[string]interface{}, error) {
+	certManagementConfig, err := certmanagement.RetrieveCertificateManagementConfig(certificateManagement)
+	if err != nil {
+		return nil, err
+	}
+
+	var (
+		clusterIssuerName = certManagementConfig.ClusterIssuerName
+		acmeConfig        = certManagementConfig.ACME
+		route53Config     = certManagementConfig.Providers.Route53
+		clouddnsConfig    = certManagementConfig.Providers.CloudDNS
+	)
+
+	var dnsProviders []certmanagement.DNSProviderConfig
+	for _, route53provider := range route53Config {
+		it := route53provider
+		dnsProviders = append(dnsProviders, &it)
+	}
+	for _, cloudDNSProvider := range clouddnsConfig {
+		it := cloudDNSProvider
+		dnsProviders = append(dnsProviders, &it)
+	}
+
+	var letsEncryptSecretName = "letsencrypt-secret"
+	providers, secrets := createDNSProviders(dnsProviders)
+
+	acmePrivateKey := acmeConfig.ACMEPrivateKey()
+	if len(acmePrivateKey) > 0 {
+		letsEncryptKeySecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      letsEncryptSecretName,
+				Namespace: common.GardenNamespace,
+			},
+			Data: map[string][]byte{
+				"tls.key": []byte(acmePrivateKey),
+			},
+		}
+		secrets = append(secrets, letsEncryptKeySecret)
+	}
+
+	err = createCertificateIssuerSecrets(k8sSeedclient, secrets)
+	if err != nil {
+		return nil, err
+	}
+
+	return map[string]interface{}{
+		"name": string(clusterIssuerName),
+		"acme": map[string]interface{}{
+			"email":               acmeConfig.Email,
+			"server":              acmeConfig.Server,
+			"privateKeySecretRef": letsEncryptSecretName,
+			"dns01": map[string]interface{}{
+				"providers": providers,
+			},
+		},
+	}, nil
+}
+
+func createCertificateIssuerSecrets(k8sSeedClient kubernetes.Client, clusterIssuerSecrets []*corev1.Secret) error {
+	for _, clusterIssuerSecret := range clusterIssuerSecrets {
+		secret, err := k8sSeedClient.CreateSecretObject(clusterIssuerSecret, true)
+		if err != nil {
+			return err
+		}
+		logger.Logger.Debugf("%s secret was created successfully", secret.Name)
+	}
+	return nil
+}
+
+func createDNSProviders(configs []certmanagement.DNSProviderConfig) ([]interface{}, []*corev1.Secret) {
```

Comment:
Split this function into one that generates secrets and one that generates helm configuration, then also the name can improve.